src/scales/ordinal.js

```diff
@@ -28,7 +28,7 @@ export function ScaleO(scale, channels, {
     if (typeof range === "function") range = range(domain);
     scale.range(range);
   }
-  return {type, domain, range, scale, hint};
+  return {type, domain, range, scale, hint, interval};
```

Comment:
I think we want to normalize the interval representation here:

```suggestion
  return {type, domain, range, scale, hint, interval: maybeInterval(interval)};
```

I know you weren’t doing that before so you could check if `typeof interval === "number"`, but I think we don’t need that anymore with my formatDefault suggestion, and it’s better to normalize input options as early as possible so we don’t have to worry about alternative representations.

Reply:
I also wanted to export it with the scale ( 
https://github.com/observablehq/plot/pull/852/files#diff-1dac81a3ab4fe192a8817bbe7a4580c5703cc70f8ccfba55f33be4aeaf556dc7 ). However, just like interpolators, it can be exported in its normalized (and opaque) form, as long as it's reusable (which is also unit-tested).